Modules/mathmodule.c

```diff
@@ -2016,6 +2016,71 @@ math_factorial(PyObject *module, PyObject *arg)
 }
 
 
+/*[clinic input]
+math.lcm
+
+    x as a: object
+    y as b: object
+    /
+least common multiple of x and y
+[clinic start generated code]*/
+
+static PyObject *
+math_lcm_impl(PyObject *module, PyObject *a, PyObject *b)
+/*[clinic end generated code: output=6f83fb6d671074ba input=bd41b785dc2a4ff1]*/
+{
+    PyObject *g, *m, *f, *ab;
+    
+    a = PyNumber_Index(a);
+        if (a == NULL) {
+            return NULL;
+    }
+    b = PyNumber_Index(b);
+    if (b == NULL) {
+        return NULL;
```

Comment:
Here the second call to `PyNumber_Index` failed. Again, we need to DECREF everything we own and then return `NULL`.

So here we have the first refcounting bug in the code: at this point, we _do_ own a reference to `a`, so it's our responsibility to DECREF it. So you need to add a `Py_DECREF(a);` before the `return NULL`.

Owned references before the return: `a`